src/oc_erchef/apps/oc_chef_wm/src/chef_wm_prometheus_json_format.erl

```diff
@@ -0,0 +1,100 @@
+-module(chef_wm_prometheus_json_format).
+
+-export([
+         content_type/0,
+         format/0,
+         format/1]).
+
+-include_lib("prometheus/include/prometheus_model.hrl").
+
+-define(PROCESS_DICT_STORAGE, chef_wm_prometheus_json_format_data).
+
+-spec content_type() -> binary().
+content_type() ->
+    <<"application/json">>.
+
+-spec format() -> binary().
+format() ->
+    format(default).
+
+-spec format(Registry :: prometheus_registry:registry()) -> binary().
+format(Registry) ->
+    put(?PROCESS_DICT_STORAGE, []),
+    Callback = fun (_, Collector) ->
+                       registry_collect_callback(Registry, Collector)
+               end,
+    prometheus_registry:collect(Registry, Callback),
+    case get(?PROCESS_DICT_STORAGE) of
+        undefined ->
+            <<"{}">>;
+        Data ->
+            %TODO(jaym) 08/27/17: It might be worthwhile making this defensive to failure... always cleanup
+            erase(?PROCESS_DICT_STORAGE),
+            jiffy:encode(Data)
+    end.
+
+registry_collect_callback(Registry, Collector) ->
+  Callback = fun (MF) ->
+                 Data = get(?PROCESS_DICT_STORAGE),
+                 put(?PROCESS_DICT_STORAGE, [mf_to_erl(MF) | Data])
+             end,
```

Comment:
It is a bit of a bummer that the API here doesn't let us pass around an accumulator instead.

Reply:
yea :(
Would have been a lot more cleaner